packages/nodes-base/nodes/Oracle/Sql/methods/listSearch.ts

```diff
@@ -0,0 +1,86 @@
+import { NodeOperationError } from 'n8n-workflow';
+import type { ILoadOptionsFunctions, INodeListSearchResult } from 'n8n-workflow';
+import * as oracleDBTypes from 'oracledb';
+
+import type { OracleDBNodeCredentials } from '../helpers/interfaces';
+import { configureOracleDB } from '../transport';
+
+export async function schemaSearch(this: ILoadOptionsFunctions): Promise<INodeListSearchResult> {
+	const credentials = await this.getCredentials<OracleDBNodeCredentials>('oracleDBApi');
+	const options = { nodeVersion: this.getNode().typeVersion };
+
+	const pool: oracleDBTypes.Pool = await configureOracleDB.call(this, credentials, options);
+
+	let conn: oracleDBTypes.Connection | undefined;
+
+	try {
+		conn = await pool.getConnection();
+
+		const response = await conn.execute<{ USERNAME: string }>(
+			'SELECT username FROM all_users',
+			[],
+			{
+				outFormat: oracleDBTypes.OUT_FORMAT_OBJECT,
+			},
+		);
+
+		const results =
+			response.rows?.map((schema) => ({
+				name: schema.USERNAME,
+				value: schema.USERNAME,
+			})) ?? [];
+
+		return { results };
+	} catch (error) {
+		throw new NodeOperationError(this.getNode(), `Failed to fetch tables: ${error.message}`);
```

Comment:
This is a function to fetch schemas, not tables
```suggestion
		throw new NodeOperationError(this.getNode(), `Failed to fetch schemas: ${error.message}`);
```